utilities/mobile_server_pool.py

```diff
@@ -0,0 +1,227 @@
+import sys
+from requests import Session
+from optparse import OptionParser
+from couchbase.bucket import Bucket
+from couchbase.n1ql import N1QLQuery
+from keywords.utils import log_info
+from couchbase.exceptions import KeyExistsError
+
+SERVER_IP = "172.23.105.177"
+USERNAME = 'Administrator'
+PASSWORD = 'esabhcuoc'
+BUCKET_NAME = "QE-mobile-pool"
+sdk_client = Bucket("couchbase://{}/{}".format(SERVER_IP, BUCKET_NAME))
+
+
+def get_nodes_from_pool_server(num_of_nodes, nodes_os_type, node_os_version, job_name_requesting_node):
+    """
+    Reserve no. of nodes from QE-mobile-pool
+    :param num_of_nodes: No. of nodes one needs
+    :param nodes_os_type: Type of OS for reserve node
+    :param node_os_version: Version of OS for reserve node
+    :param job_name_requesting_node: Name of the job requesting for node
+    :return: list of nodes reserved
+    """
+    # Getting list of available nodes through n1ql query
+    query_str = "select meta().id from `{}` where os='{}' " \
+                "AND os_version='{}' AND state='available'".format(BUCKET_NAME, nodes_os_type, node_os_version)
+    query = N1QLQuery(query_str)
+    pool_list = []
+    for row in sdk_client.n1ql_query(query):
+        doc_id = row["id"]
+        node = reserve_node(doc_id, job_name_requesting_node)
```

Comment:
reserve_node function return a boolean, my first impression of the variable name node is that reserve_node return an object (of a machine as a node). it will be nice to rename node to something more meaningful, but this is just a minor comment.

Reply:
I can return void and raise exception if nodes are not reserve. Let me know if that make sense.

Reply:
I am fine with the definition of reserve_node and its boolean return type. However, when I read this line of code (without checking the definition of reserve_node), I got a wrong impression that reserve_node returns a node object, instead of a boolean status. You may change the variable "node" to "is_node_available" or something like "nodes_reserve_status", people will not misunderstand what is doing here.